src/test/java/be/seeseemelk/mockbukkit/damage/DamageTypeMockTest.java

```diff
@@ -0,0 +1,79 @@
+			"{ 'key': 'minecraft:fireworks', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:fireball', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt_on_fire', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:unattributed_fireball', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt_on_fire', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:wither_skull', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:thrown', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:indirect_magic', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.0 }",
+			"{ 'key': 'minecraft:thorns', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:enchant.thorns.hit', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:explosion', 'damageScaling': 'ALWAYS', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:player_explosion', 'damageScaling': 'ALWAYS', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:sonic_boom', 'damageScaling': 'ALWAYS', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.0 }",
+			"{ 'key': 'minecraft:bad_respawn_point', 'damageScaling': 'ALWAYS', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'INTENTIONAL_GAME_DESIGN', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:outside_border', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.0 }",
+			"{ 'key': 'minecraft:generic_kill', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.0 }"
+	})
+	void from_GivenMinecraftExamples(String jsonString) {
+		JsonObject jsonObject = JsonParser.parseString(jsonString).getAsJsonObject();
+
+		var expectedExhaustion = jsonObject.get(DamageTypeMock.EXHAUSTION).getAsFloat();
+		var expectedDeathMessage = jsonObject.get(DamageTypeMock.DEATH_MESSAGE_TYPE).getAsString();
+		var expectedDamageScaling = jsonObject.get(DamageTypeMock.DAMAGE_SCALING).getAsString();
+		var expectedKey = jsonObject.get(DamageTypeMock.KEY).getAsString();
+		var expectedSound = jsonObject.get(DamageTypeMock.SOUND).getAsString();
```

Comment:
I did not know this existed in java (hehe). Anyhow, I believe this goes against the strength of java, I've always loved how explicit it is, so please change this to explicit types.